repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java

```diff
@@ -846,6 +859,54 @@ public String createVertexIndex(AtlasGraphManagement management, String property
         return indexFieldName;
     }
 
+    public String createVertexIndex(AtlasAttributeDef attributeDef, AtlasGraphManagement management, String propertyName, UniqueKind uniqueKind, Class propertyClass,
+                                  AtlasCardinality cardinality, boolean createCompositeIndex, boolean createCompositeIndexWithTypeAndSuperTypes, boolean isStringField, HashMap<String, Object> indexTypeESConfig, HashMap<String, HashMap<String, Object>> indexTypeESFields) {
+        String indexFieldName = null;
+
+        if (propertyName != null) {
+            AtlasPropertyKey propertyKey = management.getPropertyKey(propertyName);
+
+            if (propertyKey == null) {
+                propertyKey = management.makePropertyKey(propertyName, propertyClass, cardinality);
+            }
+            else attributeDef.setIndexPresent(true);
```

Comment:
Plz talk to @sumandas0  once, I see `setIndexPresent` is used in this method at multiple places. I feel we should set flag only once, after all conditions are satisfied.

Reply:
Flag at line 872 is to check whether an index of same name is mapped or not. If there is no index, flag will set at line 880 after creating mixed index.